apps/api/src/app/environments/usecases/construct-framework-workflow/construct-framework-workflow.usecase.ts

```diff
@@ -0,0 +1,181 @@
+
+    if (!stepTemplate) {
+      throw new NotFoundException(`Step template not found for step ${staticStep.stepId}`);
+    }
+
+    const stepType = stepTemplate.type;
+    const { stepId } = staticStep;
+
+    if (!stepId) {
+      throw new NotFoundException(`Step id not found for step ${staticStep.stepId}`);
+    }
+
+    const stepControls = stepTemplate.controls;
+
+    if (!stepControls) {
+      throw new NotFoundException(`Step controls not found for step ${staticStep.stepId}`);
+    }
+
+    switch (stepType) {
+      case StepTypeEnum.IN_APP:
+        return step.inApp(
+          // The step id is used internally by the framework to identify the step
+          stepId,
+          // The step callback function. Takes controls and returns the step outputs
+          async (controlValues) => {
+            // TODO: insert custom in-app hydration logic here.
+            return controlValues as InAppOutput;
+          },
+          // Step options
+          this.constructChannelStepOptions(staticStep)
+        );
+      case StepTypeEnum.EMAIL:
+        return step.email(
+          stepId,
+          async (controlValues) => {
+            // TODO: insert custom Maily.to hydration logic here.
+            return controlValues as EmailOutput;
+          },
+          this.constructChannelStepOptions(staticStep)
+        );
+      case StepTypeEnum.SMS:
+        return step.inApp(
+          stepId,
+          async (controlValues) => {
+            // TODO: insert custom SMS hydration logic here.
+            return controlValues as SmsOutput;
+          },
+          this.constructChannelStepOptions(staticStep)
+        );
+      case StepTypeEnum.CHAT:
+        return step.inApp(
+          stepId,
+          async (controlValues) => {
+            // TODO: insert custom chat hydration logic here.
+            return controlValues as ChatOutput;
+          },
+          this.constructChannelStepOptions(staticStep)
+        );
+      case StepTypeEnum.PUSH:
+        return step.inApp(
+          stepId,
+          async (controlValues) => {
+            // TODO: insert custom push hydration logic here.
+            return controlValues as PushOutput;
+          },
+          this.constructChannelStepOptions(staticStep)
+        );
+      case StepTypeEnum.DIGEST:
+        return step.digest(
+          stepId,
+          async (controlValues) => {
+            return controlValues as DigestOutput;
+          },
+          this.constructActionStepOptions(staticStep)
+        );
+      case StepTypeEnum.DELAY:
+        return step.delay(
+          stepId,
+          async (controlValues) => {
+            return controlValues as DelayOutput;
+          },
+          this.constructActionStepOptions(staticStep)
+        );
+      default:
+        throw new NotFoundException(`Step type ${stepType} not found`);
+    }
+  }
+
+  private constructChannelStepOptions(staticStep: NotificationStepEntity): Required<Parameters<ChannelStep>[2]> {
```

Comment:
Note, we use `Required` generic to ensure that we always support/dogfood all Novu Framework options.